src/platform/w3dengine/client/w3ddebugicons.cpp

```diff
@@ -0,0 +1,273 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief Class for drawing small debug icons on screen.
+ * Commonly used for debugging things like pathfinding, trajectories, flight paths.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3ddebugicons.h"
+#include "gamelogic.h"
+#include "globaldata.h"
+#include "vertmaterial.h"
+#include <new>
+
+#ifdef BUILD_WITH_D3D8
+#include "dx8indexbuffer.h"
+#include "dx8vertexbuffer.h"
+#include "dx8wrapper.h"
+#endif
+
+DebugIcon *W3DDebugIcons::m_debugIcons;
+int W3DDebugIcons::m_numDebugIcons;
+int W3DDebugIcons::m_numUnknown;
+
+W3DDebugIcons::W3DDebugIcons() : m_vertexMaterial(VertexMaterialClass::Get_Preset(VertexMaterialClass::PRELIT_DIFFUSE))
+{
+    Allocate_Icons_Array();
+}
+
+W3DDebugIcons::W3DDebugIcons(const W3DDebugIcons &src)
+{
+    *this = src;
+}
+
+W3DDebugIcons &W3DDebugIcons::operator=(const W3DDebugIcons &that)
+{
+    captainslog_dbgassert(0, "Cannot make a copy of W3DDebugIcons!");
```

Comment:
May I suggest to just make copy constructor and assignment operator `= delete;` ? This way we get compile error if someone tries to copy, as opposed to a runtime assert.